### PR TITLE
Added File storage controller

### PR DESCRIPTION
**Pull request by Mixer57**

### CODE REVIEW

**Review by peterliapin — CHANGES_REQUESTED** [review]

File: src/OnlineSales/Data/PgDbContext.cs [hunk 1]
```diff
@@ -204,8 +205,7 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
             optionsBuilder.UseNpgsql(
                 postgresConfig.ConnectionString,
                 b => b.MigrationsHistoryTable("_migrations"))
-            .UseSnakeCaseNamingConvention()
-            .ReplaceService<IMigrationsSqlGenerator, CustomSqlServerMigrationsSqlGenerator>();
```
[the remark this answers is not in the document]
Comment (Mixer57) [reply src/OnlineSales/Data/PgDbContext.cs:208]: Restored

File: src/OnlineSales/Controllers/FilesController.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        {
+            pathToFile = Uri.UnescapeDataString(pathToFile);
+
+            var scope = Path.GetDirectoryName(pathToFile)!.Replace(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
+            var fname = Path.GetFileName(pathToFile);
+
+            var uploadedImageData = await pgDbContext!.Files!.Where(e => e.ScopeUid == scope && e.Name == fname).FirstOrDefaultAsync();
+
+            return uploadedImageData == null
+                ? throw new EntityNotFoundException(nameof(Media), $"{pathToFile}")
```
[the remark this answers is not in the document]
Comment (Mixer57) [reply src/OnlineSales/Controllers/FilesController.cs]: fixed

**Review by IgorKostyukov — APPROVED** [review]

File: src/OnlineSales/Entities/File.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+// Licensed under the MIT license. See LICENSE file in the samples root for full license information.
+// </copyright>
+
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using OnlineSales.DataAnnotations;
+
+namespace OnlineSales.Entities
+{
+    [Table("files")]
```
Comment (peterliapin) [inline src/OnlineSales/Entities/File.cs:11]: @Mixer57 table name must be in singular form, i.e. "file"

File: src/OnlineSales/Migrations/20230703042743_DbFileStorage.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+namespace OnlineSales.Migrations
+{
+    /// <inheritdoc />
+    public partial class DbFileStorage : Migration
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateTable(
+                name: "files",
```
Comment (peterliapin) [inline src/OnlineSales/Migrations/20230703042743_DbFileStorage.cs:16]: @Mixer57 table name must be in singular form, i.e. "file"

**Review by peterliapin — CHANGES_REQUESTED** [review]

File: src/OnlineSales/Entities/File.cs (hunk 3 above)
Comment (Mixer57) [reply src/OnlineSales/Entities/File.cs]: fixed

File: src/OnlineSales/Migrations/20230703042743_DbFileStorage.cs (hunk 4 above)
Comment (Mixer57) [reply src/OnlineSales/Migrations/20230703042743_DbFileStorage.cs]: fixed

**Pull request merged**